src/mrtimer.rs

```diff
@@ -0,0 +1,146 @@
+//! API for the MRT peripheral
+//!
+//! Please be aware that this doesn't try to abstract everything, it only
+//! implements the embedded-hal `Timer` functionality.
+//!
+//! The MRT consists of 4 channels, which are mostly separate and can each act
+//! as a run-of-the-mill timer.
+//!
+//! # Example
+//!
+//! ``` no_run
+//! use lpc8xx_hal::prelude::*;
+//! use lpc8xx_hal::Peripherals;
+//!
+//! use nb::block;
+//!
+//! let mut p = Peripherals::take().unwrap();
+//!
+//! let mut syscon = p.SYSCON.split();
+//! let [mut timer, _, _, _] = p.SYST.enable_delay();
+//! timer.start(12_000_000u32);
+//! loop {
+//!     block!(timer.wait()).unwrap();
+//! }
+//! ```
+
+use crate::{
+    pac::{mrt0::CHANNEL, MRT0},
+    reg_proxy::RegProxy,
+    syscon,
+};
+
+use embedded_hal::timer::{CountDown, Periodic};
+use nb::{Error, Result};
+use void::Void;
+
+/// Represent a MRT0 instance
+pub struct MRTimer {
```

Comment:
I'd prefer the name `MRT` (and `mrt` for the module), as that makes is consistent with the user manual, making it clear what peripheral this API is for. It's also what we do for other peripherals.

Reply:
Sure, why not